packages/docs/docs/contributing/troubleshooting.md

```diff
@@ -0,0 +1,224 @@
+
+### Issue: Cannot resolve imports
+
+**Solution:**
+
+1. Check `tsconfig.json` for path mappings
+2. Check package.json `exports` field (especially for loot-core)
+3. Verify platform-specific imports (`.web`, `.electron`, `.api`)
+4. Use absolute imports in `desktop-client` (enforced by ESLint)
+5. Ensure you're importing from the correct package path
+
+### Issue: Platform-specific import errors
+
+**Solution:**
+
+- Don't directly reference platform-specific imports (`.api`, `.web`, `.electron`)
+- Use conditional exports in `loot-core` for platform-specific code
+- Platform resolution happens at build time via package.json exports
+
+## Build Failures
+
+### Issue: Build fails with errors
+
+**Solution:**
+
+1. Clean build artifacts:
+   ```bash
+   rm -rf packages/*/dist packages/*/lib-dist packages/*/build
+   ```
+2. Reinstall dependencies:
+   ```bash
+   yarn install
+   ```
+3. Check Node.js version (requires >=22):
+   ```bash
+   node --version
+   ```
+4. Check Yarn version (requires ^4.9.1):
+   ```bash
+   yarn --version
+   ```
+
+### Issue: Native module build failures (better-sqlite3)
+
+**Solution:**
+
+1. On Windows: Ensure you selected "Automatically install the necessary tools" during Node.js installation
+2. Run `yarn rebuild-electron` for Electron builds
+3. Run `yarn workspace loot-core rebuild` for Node.js builds
+4. Ensure you have the necessary build tools installed (Python, Visual Studio Build Tools on Windows)
+
+## Development Server Issues
+
+### Issue: Development server won't start
+
+**Solution:**
+
+1. Check if port is already in use
+2. Ensure all dependencies are installed: `yarn install`
+3. Try clearing node_modules and reinstalling:
+   ```bash
+   rm -rf node_modules packages/*/node_modules
+   yarn install
+   ```
+4. Check for error messages in the console
+
+### Issue: Hot reload not working
+
+**Solution:**
+
+1. Ensure you're running the correct development command
+2. Check if file watchers are working (may be limited on some systems)
+3. Try restarting the development server
+4. Check for file system permission issues
+
+## Workspace Command Issues
+
+### Issue: Workspace command not found
+
+**Solution:**
+
+1. Ensure you're running commands from the root directory
+2. Verify workspace name is correct: `yarn workspaces list`
+3. Check package.json for available scripts
+4. Use correct workspace alias (e.g., `@actual-app/web` instead of `desktop-client`)
+
+## Git Issues
+
+### Issue: Pre-commit hooks failing
+
+**Solution:**
+
+1. Ensure Husky is set up: `yarn prepare`
+2. Run linting manually: `yarn lint:fix`
+3. Run type checking: `yarn typecheck`
+4. Fix any errors before committing
+
+### Issue: Merge conflicts
+
+**Solution:**
+
+1. Keep your branch up to date with master:
+   ```bash
+   git fetch origin
+   git rebase origin/master
+   ```
+2. Resolve conflicts carefully
+3. Run `yarn typecheck` and `yarn lint:fix` after resolving conflicts
+4. Run tests to ensure everything still works
```

Comment:
IMO this is too complex to document here, merge conflicts are documented extensively by others and google will be a better tool.